### ISSUE TITLE
Can't ban calls to window or document

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#4613]
### Bug Report

- __TSLint version__: 3.2.4
- __TypeScript version__: 5.14.0
- __Running TSLint via__: CLI & VSCode

#### TypeScript code being linted

```ts
redirect(): void {
  window.location.href = this.logoutUrl;
}
```

I've tried 2 different `tslint.json` configurations:

```json
{"name": "window", "message": "Don't call `window` directly, instead inject the Angular `Window` provider and call `this.window`"},
{"name": "document", "message": "Don't call `document` directly, instead inject the Angular `Window` provider and call `this.window.document`"},
```
or
```json
{"name": ["window", "*"], "message": "Don't call `window` directly, instead inject the Angular `Window` provider and call `this.window`"},
{"name": ["document", "*"], "message": "Don't call `document` directly, instead inject the Angular `Window` provider and call `this.window.document`"},
```

#### Actual behavior
`All files pass linting.`

#### Expected behavior
Expected errors where `window.location.href` is called

**Comment by JoshuaKGoldberg** [comment]
Good news, @chrismbarr! You can use [`no-restricted-globals`](https://palantir.github.io/tslint/rules/no-restricted-globals) to get around this now. It's targeted more for your use case, I think.
